src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java

```diff
@@ -192,6 +193,9 @@ public void performMining() {
                 this.wasActiveAndNeedsUpdate = true;
                 this.setActive(false);
             }
+        } else {
+            // we drain the energy at the end here to ensure that we only drain if a block was found
+            drainStorages(false);
```

Comment:
This is wrong place where to drain. There are 2 reasons: 
1) logically it does not belong here as the work was done above
2) in case that we would mine last block in current batch `blocksToMine.isEmpty()` will be `true` and this would not be called

So please move it directly after `mineAndInsertItems()`.

Reply:
I believe moving it after `mineAndInsertItems()` is the wrong location, as then it would only drain every mining operation instead of every tick, after discussions in the discord it was decided that every tick was acceptable.

Reply:
I was not part of that discussion and decision seems as regression to me. Because it would eat energy even when the miner is inactive (specifically when it's inventory is full).
So I won't approve it. Feel free to contact devs, that were part of discussion to take over.

Reply:
Thanks for identifying the bug I created, fixed that and reverted functionality to how it was before changes.